tools/glide-pindeps/glide/glide.go

```diff
@@ -0,0 +1,60 @@
+package glide
+
+import (
+	"bytes"
+	"fmt"
+	"time"
+
+	yaml "gopkg.in/yaml.v2"
+)
+
+type LockFile struct {
+	Hash    string    `yaml:"hash"`
+	Updated time.Time `yaml:"updated"`
+
+	Imports []*LockFileImport `yaml:"imports"`
+}
+
+func (l *LockFile) Decode(b []byte) error {
+	return yaml.Unmarshal(b, l)
+}
+
+type YamlFile struct {
+	raw []byte
+
+	Package     string            `yaml:"package"`
+	ExcludeDirs []string          `yaml:"excludeDirs"`
+	Imports     []*YamlFileImport `yaml:"import"`
+}
+
+// AppendAndEncodeRaw appends a slice of YamlFileImport to the original
+// raw file data, and returns the encoded result
+func (y *YamlFile) AppendAndEncodeRaw(newImports []*YamlFileImport) ([]byte, error) {
```

Comment:
This method name smells bad and I don't like it hanging off of the struct involved.  I don't think I care what I append it to or even care to know that I'm appending it.  

I think where this is used, you get the original file content, write it to the output stream, then serialize your imports and append them to the same stream and call it a day.